docs/content/sips/000-spin-test.md

```diff
@@ -0,0 +1,190 @@
+    /// interfaces are configured through that combination.
+    export set-component-id: func(component-id: string);
+    import get-manifest: func() -> string;
+
+	/// A way to say that a certain URL is associated with a response
+    export http-handler;
+    /// How the calls to the kv store are tracked
+    export key-value-calls;
+    
+}
+
+interface http-handler {
+    use wasi:http/types@0.2.0.{future-incoming-response};
+    set-response: func(url: string, response: future-incoming-response);
+}
+
+interface key-value-calls {
+    reset: func();
+    get: func() -> list<tuple<string, list<get-call>>>;
+    set: func() -> list<tuple<string, list<set-call>>>;
+
+    record get-call {
+        key: string
+    }
+
+    record set-call {
+        key: string,
+        value: list<u8>
+    }
+}
+```
+
+### The `fermyoin:spin-test/runner` world
+
+```
+world runner {
+	/// The host must supply a manifest
+    import get-manifest: func() -> string;
+    /// The host must give the runner the ability to create HTTP resources
+    import http-helper;
+    /// The host can then run the test
+    export run: func();
+}
+```
+
+## Open Questions
+
+- Should `spin-test` be a stand alone CLI experience, a plugin for `spin` CLI, or both?
+- How do we handle Spin applications with more than component?
+    - There’s no way to compose a statically defined component with N number of components. This means the router component must know how many components it will be composed with ahead of time, and so the reasonable number of 1 is picked.
+    - We could work around this by moving the router out of a component and into the host, but this makes the host smarter which would like to avoid. We could also potentially synthesize a router component based on the `spin.toml` manifest instead of relying on a statically defined one, but this is certainly not trivial.
+- How should test discovery work?
+    - The most straightforward way would be for `spin-test` to require the user to specify the path to a `spin-test` compliant test component binary.
+    - Another way would be for there to be a convention around a `tests` directory. This `tests` directory could include both built `spin-test` compliant test component binaries as well as directories with source code and a `spin-test.toml` configuration file that specifies how the source code is built into a `spin-test` compliant test component binary. `spin-test` would then build all the tests that need to be built and run each test binary.
+    - We could also carve away space in the `spin.toml` manifest (e.g., by using the `[component.<id>.tool]` section). Such a solution might be more appropriate if we determine that users will normally want very few test component binaries. If, however, it proves to be useful to potentially have many test binaries, splitting the configuration out to each test binary in the form of a bespoke `spin-test.toml` per test binary might be the better solution.
```

Comment:
I suppose we could also do both (eventually), with a compatible schema reused between `spin.toml` for one-test-per-component and `tests/spin-test.toml` for more complex scenarios.

Reply:
Another option to consider might be the other `tool` section in `spin.toml`, e.g. `[application.tool.spin-test.test-a]`. I'm not sure I would actually _like_ that but it would avoid having another config file.